.github/workflows/PINP-test.yaml

```diff
@@ -22,14 +22,35 @@ jobs:
         sudo apt install -y podman
         podman --version
 
+    # garbage needed because otherwise the cache fails to save properly. Suggested by https://github.com/containers/podman/discussions/17868
+    - name: Special tar sauce
+      run: |
+        sudo mv -fv /usr/bin/tar /usr/bin/tar.orig
+        echo -e '#!/bin/sh\n\nsudo /usr/bin/tar.orig --xattrs --xattrs-include='"'"'*'"'"' "$@"' | sudo tee -a /usr/bin/tar
+        sudo chmod +x /usr/bin/tar
+
+    - uses: actions/cache@v4
+      with:
+        path: |
+          ~/.local/share/containers
+        key: podman-in-docker-build-cache
+
```

Comment:
why do we need podman in docker and podman in podman caches?

Reply:
Because the data stored in them is completely independent. One caches the layers of building the container that orchestrates the build, up, test process that changes less frequently, and the other is the cache of the podman state within that container to make the podman build part of the CI run faster. I think the majority of the speedup comes from the second one, but together they half the time that CI takes to run from 2 minutes to 1 minute if the cache is hit a hit.